pkg/crane/rebase.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2021 Google LLC All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package crane
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/mutate"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+)
+
+const (
+	baseDigestAnnotation = "org.opencontainers.image.base.digest"
+	baseRefAnnotation    = "org.opencontainers.image.base.ref.name"
+)
+
+// Rebase parses the references and uses them to perform a rebase.
+//
+// If oldBase or newBase are "", Rebase attempts to derive them using
+// annotations in the original image. If those annotations are not found,
+// Rebase returns an error.
+//
+// If rebasing is successful, base image annotations are set on the resulting
+// image to facilitate implicit rebasing next time.
+func Rebase(orig, oldBase, newBase string, opt ...Option) (v1.Image, error) {
```

Comment:
Given that this returns a `v1.Image`, I think `orig` should also be a `v1.Image`.

Alternatively, this could operate on just refs, so that we could make it work for index as well.

Reply:
I'd also be fine just moving all of `pkg/crane.Rebase` back into cmd/crane/cmd/rebase.go for now, at least until we hammer out index rebasing in #1086, then see how we want `pkg/crane.Rebase` to look and work.